tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Matter/MatterRepository.cs

```diff
@@ -1514,24 +1519,193 @@ internal string GetStampPropertyValue(Dictionary<string, object> stampedProperty
         /// <returns>Users that can be stamped</returns>
         private string GetMatterAssignedUsers(Matter matter)
         {
-            string currentUsers = string.Empty;
-            string separator = string.Empty;
-            if (null != matter && 0 < matter.AssignUserNames.Count)
-            {
-                foreach (IList<string> userNames in matter.AssignUserNames)
+            try
+            {
+                string currentUsers = string.Empty;
+                string separator = string.Empty;
+                if (null != matter && 0 < matter.AssignUserNames.Count)
+                {
+                    foreach (IList<string> userNames in matter.AssignUserNames)
+                    {
+                        currentUsers += separator + string.Join(ServiceConstants.SEMICOLON, userNames.Where(user => !string.IsNullOrWhiteSpace(user)));
+                        separator = ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR;
+                    }
+                }
+                return currentUsers;
+            }catch(Exception ex)
+            {
+                customLogger.LogError(ex, MethodBase.GetCurrentMethod().DeclaringType.Name, MethodBase.GetCurrentMethod().Name, logTables.SPOLogTable);
+                throw;
+            }
+        }
+
+        public bool SetPermission(ClientContext clientContext, IList<IList<string>> assignUserName, IList<string> permissions, string listName) => spList.SetPermission(clientContext, assignUserName, permissions, listName);
+
+        public GenericResponseVM ShareMatterToExternalUser(MatterInformationVM matterInformation) => extrnalSharing.ShareMatter(matterInformation);
+
+        public bool OneNoteUrlExists(MatterInformationVM matterInformation) => search.PageExists(matterInformation.Client, matterInformation.RequestedUrl);
+
+        /// <summary>
+        /// This method will get all the Field columns of the content type as the JSON object which is used to render the dynamic UI.
+        /// </summary>
+        /// <param name="contentTypeName"></param>
+        /// <param name="client"></param>
+        /// <returns>Json</returns>
+        public string GetMatterProvisionExtraProperties(string contentTypeName, Client client)
```

Comment:
New method to get the sitecolumns with metada of new contenttype 'Additional Matter Properties' associated with the term store.